src/Combine.cc

```diff
@@ -1269,9 +1265,8 @@ void Combine::addDiscreteNuisances(RooWorkspace *w){
     while (RooAbsArg *arg = (RooAbsArg*)it->Next()) {
       (CascadeMinimizerGlobalConfigs::O().allRooMultiPdfs).add(*(dynamic_cast<RooMultiPdf*>(arg)));
       RooAbsPdf *pdf = dynamic_cast<RooAbsPdf*>(arg);
-      RooArgSet *pdfPars = pdf->getParameters((const RooArgSet*)0);
-      std::unique_ptr<TIterator> iter_v(pdfPars->createIterator());
-      for (RooAbsArg *a = (RooAbsArg *) iter_v->Next(); a != 0; a = (RooAbsArg *) iter_v->Next()) {
+      std::unique_ptr<RooArgSet> pdfPars{pdf->getParameters((const RooArgSet*)0)};
```

Comment:
A memory leak is fixed here. Note that `getParameters` returns an owning pointer that needs to be deleted.